src/orchestrator.ts

```diff
@@ -200,40 +231,214 @@ export class Orchestrator implements Callbacks {
+        } else {
+            this.logger.error(
+                `Expected to find state with sequence number '${msg.globalSequenceNumber}', but did not. Has it already been handled?`,
+            )
+        }
+    }
+
+    async startStreamMessage(
+        streamIdentify: StreamIdentify,
+        sendingStream: AsyncIterable<StreamChunk> &
+            grpc.ServerDuplexStream<StreamChunk, ReceivingStreamControl>,
+    ): Promise<number> {
+        const globalSequenceNumber = this.globalSequenceNumber++
+        const writeToSender = promisify(sendingStream.write.bind(sendingStream))
+
+        const sourceRunner = this.instantiators[streamIdentify.runner]
+        if (!sourceRunner) {
+            throw (
+                'Failed to find correct source runner with uri ' +
+                streamIdentify.runner
+            )
+        }
+
+        const targetInstantiator =
+            this.channelToInstantiator[streamIdentify.channel]
+
+        // If the instantiator exists, it will send a processed message
+        // When this happens, the onMessageProcessedCb will be called
+        // If the instantiator does not exist, it cannot send a processed message
+        // So the onMessageProcessedCb should be called when the stream is closed on the writer's side
+        if (targetInstantiator) {
+            this.runningMessages[globalSequenceNumber] =
+                sourceRunner.onMessageProcessedCb(
+                    streamIdentify.localSequenceNumber,
+                    streamIdentify.channel,
+                )
+
+            const readerConnected = new Promise<ReceivingStream>((res) => {
+                this.waitForConnectingReceivingStream[globalSequenceNumber] =
+                    res
+            })
+
+            targetInstantiator.streamMessage({
+                channel: streamIdentify.channel,
+                globalSequenceNumber,
+            })
+
+            const receivingStream = await readerConnected
+            const writeToReceiver = promisify(
+                receivingStream.write.bind(receivingStream),
+            )
+            const log = this.logLevels[streamIdentify.channel]
+
+            sendingStream.on('data', async (chunk) => {
+                if (chunk.data !== undefined) {
+                    if (log) {
+                        log(decoder.decode(chunk.data!.data))
+                    }
+                    await writeToReceiver(chunk.data)
+                }
+            })
+
+            sendingStream.on('end', () => {
+                try {
+                    receivingStream.end()
+                } catch (ex) {
+                    if (ex instanceof Error) {
+                        console.error(ex.stack)
+                    }
+                }
+            })
+
+            receivingStream.on('data', async (d) => {
+                try {
+                    await writeToSender(d)
+                } catch (ex) {
+                    console.log('Error happend', ex)
```

Comment:
```suggestion
                    console.log('Error happened', ex)
```
Could this (and the above console.error) be a log statement through the logging system?